rcl_action/test/rcl_action/test_wait.cpp

```diff
@@ -0,0 +1,847 @@
+    rcl_ret_t ret;
+    rcl_init_options_t init_options = rcl_get_zero_initialized_init_options();
+    ret = rcl_init_options_init(&init_options, allocator);
+    ASSERT_EQ(RCL_RET_OK, ret) << rcl_get_error_string().str;
+    OSRF_TESTING_TOOLS_CPP_SCOPE_EXIT(
+    {
+      EXPECT_EQ(RCL_RET_OK, rcl_init_options_fini(&init_options)) << rcl_get_error_string().str;
+    });
+    context = rcl_get_zero_initialized_context();
+    ret = rcl_init(0, nullptr, &init_options, &context);
+    ASSERT_EQ(RCL_RET_OK, ret) << rcl_get_error_string().str;
+    this->node = rcl_get_zero_initialized_node();
+    rcl_node_options_t node_options = rcl_node_get_default_options();
+    ret = rcl_node_init(&this->node, "test_action_server_node", "", &context, &node_options);
+    ASSERT_EQ(RCL_RET_OK, ret) << rcl_get_error_string().str;
+    ret = rcl_clock_init(RCL_ROS_TIME, &this->clock, &allocator);
+    ASSERT_EQ(RCL_RET_OK, ret) << rcl_get_error_string().str;
+    const rosidl_action_type_support_t * ts = ROSIDL_GET_ACTION_TYPE_SUPPORT(
+      test_msgs, Fibonacci);
+    const rcl_action_server_options_t options = rcl_action_server_get_default_options();
+    const char * action_name = "test_action_server_name";
+    this->action_server = rcl_action_get_zero_initialized_server();
+    ret = rcl_action_server_init(
+      &this->action_server, &this->node, &this->clock, ts, action_name, &options);
+    ASSERT_EQ(ret, RCL_RET_OK) << rcl_get_error_string().str;
+    EXPECT_FALSE(rcl_error_is_set()) << rcl_get_error_string().str;
+  }
+
+  void TearDown() override
+  {
+    // Finalize
+    rcl_ret_t ret = rcl_action_server_fini(&this->action_server, &this->node);
+    EXPECT_EQ(ret, RCL_RET_OK) << rcl_get_error_string().str;
+    ret = rcl_clock_fini(&this->clock);
+    EXPECT_EQ(ret, RCL_RET_OK) << rcl_get_error_string().str;
+    ret = rcl_node_fini(&this->node);
+    EXPECT_EQ(ret, RCL_RET_OK) << rcl_get_error_string().str;
+    ret = rcl_shutdown(&context);
+    EXPECT_EQ(RCL_RET_OK, ret) << rcl_get_error_string().str;
+    ret = rcl_context_fini(&this->context);
+    EXPECT_EQ(RCL_RET_OK, ret) << rcl_get_error_string().str;
+  }
+
+  void init_test_uuid0(uint8_t * uuid)
+  {
+    for (uint8_t i = 0; i < UUID_SIZE; ++i) {
+      uuid[i] = i;
+    }
+  }
+
+  void init_test_uuid1(uint8_t * uuid)
+  {
+    for (uint8_t i = 0; i < UUID_SIZE; ++i) {
+      uuid[i] = 15 - i;
+    }
+  }
+
+  rcl_action_server_t action_server;
+  rcl_context_t context;
+  rcl_node_t node;
+  rcl_clock_t clock;
+};  // class TestActionServer
+
+TEST_F(TestActionClientWait, test_wait_set_add_action_client) {
+  const char * action_name = "test_action_client_name";
+  const rosidl_action_type_support_t * action_typesupport =
+    ROSIDL_GET_ACTION_TYPE_SUPPORT(test_msgs, Fibonacci);
+  const rcl_action_client_options_t action_client_options =
+    rcl_action_client_get_default_options();
+
+  rcl_action_client_t action_client = rcl_action_get_zero_initialized_client();
```

Comment:
Can we move the init/fini of the action client to the test fixture (like we did for the server)?